packages/opentelemetry-instrumentation-openai/tests/traces/test_legacy_api_response.py

```
@@ -0,0 +1,289 @@
+
+def test_original_stream_integrity():
+    """Test that the original LegacyAPIResponse is not mutated"""
+    from opentelemetry.instrumentation.openai.shared import _is_legacy_api_response
+    
+    # Create a mock LegacyAPIResponse
+    mock_stream = MagicMock()
+    legacy_response = MockLegacyAPIResponse(mock_stream)
+    original_type = type(legacy_response)
+    
+    # Verify it's detected correctly
+    assert _is_legacy_api_response(legacy_response) == True
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `_is_legacy_api_response(legacy_response):` for truth checks

Replace with `_is_legacy_api_response(legacy_response)`

(E712)